### DISCUSSION TITLE
Can we invalidate all queries when loading the cache from persistent storage?

### DISCUSSION THREAD

**Discussion by SmitMaruti** [TanStack/query#5207] [Q&A] [answered]
Is it possible to invalidate all the fresh queries when loading them from persistent storage?
I have tried using `queryClient.invalidateQueries` on the application mount,  but the queries are still fresh even after mounting the actual component once the invalidateQueries promise is resolved.
FYI, I am using idb-keyval async storage for persister and the version of react-query I am using is `3.39.3`.

**Comment by TkDodo** [comment, marked as answer]
use the `PersistQueryClientProvider` in v4, which has an `onSuccess` callback that gets fired after successful restore from persistent storage. There, you can invalidate if you want.

**Reply by SmitMaruti** [reply]
Thank you. Didn't realize queries were loaded asynchronously.
